impeller/entity/contents/content_context.cc

```diff
@@ -604,4 +606,70 @@ void ContentContext::FlushCommandBuffers() const {
   GetContext()->GetCommandQueue()->Submit(buffers);
 }
 
+void ContentContext::AdditionalSetup() const {
+  // Initialize commonly used shaders that aren't defaults.
+  auto options = ContentContextOptions{
+      .sample_count = SampleCount::kCount4,
+      .color_attachment_pixel_format =
+          context_->GetCapabilities()->GetDefaultColorFormat()};
+
+  for (const auto mode : {BlendMode::kSource, BlendMode::kSourceOver}) {
+    for (const auto geometry :
+         {PrimitiveType::kTriangle, PrimitiveType::kTriangleStrip}) {
+      options.blend_mode = mode;
+      options.primitive_type = geometry;
+      GetSolidFillPipeline(options);
+      if (context_->GetCapabilities()->SupportsSSBO()) {
+        GetLinearGradientSSBOFillPipeline(options);
+        GetRadialGradientSSBOFillPipeline(options);
+        GetSweepGradientSSBOFillPipeline(options);
+        GetConicalGradientSSBOFillPipeline(options);
+        GetTexturePipeline(options);
+      }
+    }
+  }
+
+  options.blend_mode = BlendMode::kDestination;
+  options.primitive_type = PrimitiveType::kTriangleStrip;
+  for (const auto stencil_mode :
+       {ContentContextOptions::StencilMode::kLegacyClipIncrement,
+        ContentContextOptions::StencilMode::kLegacyClipDecrement,
+        ContentContextOptions::StencilMode::kLegacyClipRestore}) {
+    options.stencil_mode = stencil_mode;
+    GetClipPipeline(options);
+  }
+
+  std::string vendor = GetContext()->DescribeGpuModel();
```

Comment:
Do we imagine more changes like this in the future?

1. Does it makes sense to have this in a backend-specific context thingy?
2. Should there be a TODO with what we'd do if more of these cases are found later?

Reply:
1. No, because the backend context doesn't have access to any of the shader caches, that is all in ContentContext. I could move the texture/rp creation stuff to context_vk, but then we'd have different setup methods in different places, and I kind of think its better to keep them together. I could see either case.

2. I dunno, that feels speculative. If we find more weird cases we'll figure it out then?

Reply:
Can we leave this just as a comment and run the logic unconditionally?

It'll be difficult to keep track of where we have different vendor based driver hacks - if we're going to warm up like this why not just warm it up everywhere?